### PR TITLE
Add dependency for read op in standalone executor

### PR DESCRIPTION
**Pull request by From00**
### PR types
New features

### PR changes
Others

### Describe

Add dependency for read op in standalone executor, equivalent to add_reader_dependency_pass in PR https://github.com/PaddlePaddle/Paddle/pull/23301

### CODE REVIEW

**Review by zhiqiu — COMMENTED** [review]
do not use static library

File: paddle/fluid/framework/new_executor/interpreter/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+cc_library(
+  add_dependency
```
Comment (zhiqiu) [inline paddle/fluid/framework/new_executor/interpreter/CMakeLists.txt]: Better rename as noun

Comment (From00) [reply paddle/fluid/framework/new_executor/interpreter/CMakeLists.txt]: Done

**Review by zhwesky2010 — APPROVED** [review]
LGTM for test_add_reader_dependency failed

**Review by zhiqiu — APPROVED** [review]
LGTM

**Review by XieYunshen — APPROVED** [review]
LGTM for `set_tests_properties(test_add_reader_dependency_for_interpretercore
                     PROPERTIES TIMEOUT 120)`

**Pull request merged**